http_check/datadog_checks/http_check/http_check.py

```diff
@@ -38,43 +35,50 @@ class HTTPCheck(NetworkCheck):
     SC_STATUS = 'http.can_connect'
     SC_SSL_CERT = 'http.ssl_cert'
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        NetworkCheck.__init__(self, name, init_config, agentConfig, instances)
+    HTTP_CONFIG_REMAPPER = {
+        'client_cert': {'name': 'tls_cert'},
+        'client_key': {'name': 'tls_private_key'},
+        'disable_ssl_validation': {'name': 'tls_verify', 'invert': True, 'default': True},
+        'ignore_ssl_warning': {'name': 'tls_ignore_warning'},
+        'ca_certs': {'name': 'tls_ca_cert'},
+    }
+
+    def __init__(self, name, init_config, instances):
+        super(HTTPCheck, self).__init__(name, init_config, instances)
 
         self.ca_certs = init_config.get('ca_certs')
         if not self.ca_certs:
             self.ca_certs = get_ca_certs_path()
 
+        self.HTTP_CONFIG_REMAPPER['ca_certs']['default'] = self.ca_certs
+
+        if is_affirmative(self.instance.get('disable_ssl_validation', True)):
+            self.http.options['verify'] = False
+
     def _check(self, instance):
         (
             addr,
-            ntlm_domain,
-            username,
-            password,
             client_cert,
             client_key,
             method,
             data,
             http_response_status_code,
-            timeout,
             include_content,
             headers,
             response_time,
             content_match,
             reverse_content_match,
             tags,
-            disable_ssl_validation,
             ssl_expire,
             instance_ca_certs,
             weakcipher,
             check_hostname,
-            ignore_ssl_warning,
-            skip_proxy,
             allow_redirects,
             stream,
         ) = from_instance(instance, self.ca_certs)
-
+        timeout = self.http.options['timeout'][0]
         start = time.time()
+        self.http.options['headers'] = headers
```

Comment:
What I understand is that the http_check allows *not* to send the default agent headers with the request and thus we have to manually override it, am I right?
In that case I also think a comment could help :)